File: software/examples/php/ExampleSimple.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickIMU.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickIMU;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XXYYZZ'; // Change XXYYZZ to the UID of your IMU Brick

$ipcon = new IPConnection(); // Create IP connection
$imu = new BrickIMU(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get current quaternion
$quaternion = $imu->getQuaternion();

echo "Quaternion[X]: " . $quaternion['x'] . "\n";
echo "Quaternion[Y]: " . $quaternion['y'] . "\n";
echo "Quaternion[Z]: " . $quaternion['z'] . "\n";
echo "Quaternion[W]: " . $quaternion['w'] . "\n";

echo "Press key to exit\n";
fgetc(fopen('php://stdin', 'r'));
$ipcon->disconnect();

?>

File: software/examples/python/example_simple.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XXYYZZ" # Change XXYYZZ to the UID of your IMU Brick

from tinkerforge.ip_connection import IPConnection
from tinkerforge.brick_imu import BrickIMU

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    imu = BrickIMU(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Get current quaternion
    x, y, z, w = imu.get_quaternion()

    print("Quaternion [X]: " + str(x))
    print("Quaternion [Y]: " + str(y))
    print("Quaternion [Z]: " + str(z))
    print("Quaternion [W]: " + str(w))

    raw_input("Press key to exit\n") # Use input() in Python 3
    ipcon.disconnect()
